src/main/java/ac/grim/grimac/utils/math/GrimMath.java

```diff
@@ -105,4 +105,10 @@ public static boolean inRange(double value, double min, double max) {
     public static boolean isNearlySame(double a, double b, double epoch) {
         return Math.abs(a-b) < epoch;
     }
+
+    public static long hashCode(double x, int y, double z) {
+        long l = (long)(x * 3129871) ^ (long)z * 116129781L ^ (long)y;
+        l = l * l * 42317861L + l * 11L;
+        return l >> 16;
```

Comment:
This should return the same result (test this)
You should also rename `l`
```suggestion
        long l = x * 3129871 ^ z * 116129781L ^ y;
        return l * l * 42317861 + l * 11 >> 16;
```

Reply:
I left it in original form to make it easier to understand. If that's not desired I will make this as short and efificent as possible.

This will also not work. You need to cast to long for ^